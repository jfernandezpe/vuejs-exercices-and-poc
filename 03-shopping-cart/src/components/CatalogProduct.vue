<template>
  <article class="card catalog-product">
    <h3>{{ name }}</h3>
    <img :src="image" width="300" />
    <p>{{ price }}</p>
    <button class="button" @click="addToCartButtonClicked" :disabled="disabled">
      Add to cart
    </button>
  </article>
</template>

<script>
export default {
  name: "CatalogProduct",
  props: {
    name: String,
    image: String,
    price: Number,
    disabled: {
      type: Boolean,
      default: false
    }
  },
  methods: {
    addToCartButtonClicked() {
      if (!this.disabled) {
        this.$emit("addToCart");
      }
    }
  }
};
</script>

<style scoped lang="scss">
article {
  display: fex;
  flex-direction: column;
}
</style>
