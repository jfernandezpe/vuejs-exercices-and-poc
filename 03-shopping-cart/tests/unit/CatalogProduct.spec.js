import { shallowMount } from "@vue/test-utils";
import CatalogProduct from "@/components/CatalogProduct.vue";

describe("CatalogProduct.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(CatalogProduct, {
      propsData: {
        name: "Awesome phone",
        image: "http://cdn.local/awesome-phone.jpg",
        price: 299.95
      }
    });
  });
  it("should render the layout", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  describe("when the add to cart button is clicked", () => {
    it("it should dispatch the add to cart event", async () => {
      wrapper.find("button").trigger("click");

      await wrapper.vm.$nextTick();
      expect(wrapper.emitted().addToCart).toBeTruthy();
    });
    describe("when the product is disabled", () => {
      it("should not dispatch add to cart event", async () => {
        wrapper.setProps({ disabled: true });
        wrapper.find("button").trigger("click");

        await wrapper.vm.$nextTick();
        expect(wrapper.emitted().addToCart).toBeFalsy();
      });
    });
  });
});
