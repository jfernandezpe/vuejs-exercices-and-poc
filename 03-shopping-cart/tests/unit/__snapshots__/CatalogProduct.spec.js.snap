// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogProduct.vue should render the layout 1`] = `
<article
  class="card catalog-product"
>
  <h3>
    Awesome phone
  </h3>
   
  <img
    src="http://cdn.local/awesome-phone.jpg"
    width="300"
  />
   
  <p>
    299.95
  </p>
   
  <button
    class="button"
  >
    Add to cart
  </button>
</article>
`;
